pkg/hintrunner/zero/zerohint_sha256.go

```diff
@@ -88,6 +94,125 @@ func createPackedSha256Hinter(resolver hintReferenceResolver) (hinter.Hinter, er
 	return newPackedSha256Hint(sha256Start, output), nil
 }
 
+// Sha256Chunk hint computes the sha2 compress operation on some input using a given state
+// and writes the new state at `output` address
+//
+// `newSha256ChunkHint` takes 3 operanders as arguments
+//   - `sha256Start` is the address where to start reading the input for `sha2_compress_function`
+//   - `state` is the current that will be passed to compute the next state
+//   - `output` is the address where to write the new state in memory
+func newSha256ChunkHint(sha256Start, state, output hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "Sha256Chunk",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_sha256.sha256_utils import (
+			//> 	compute_message_schedule, sha2_compress_function)
+			//>
+			//> _sha256_input_chunk_size_felts = int(ids.SHA256_INPUT_CHUNK_SIZE_FELTS)
+			//> assert 0 <= _sha256_input_chunk_size_felts < 100
+			//> _sha256_state_size_felts = int(ids.SHA256_STATE_SIZE_FELTS)
+			//> assert 0 <= _sha256_state_size_felts < 100
+			//> w = compute_message_schedule(memory.get_range(
+			//> 	ids.sha256_start, _sha256_input_chunk_size_felts))
+			//> new_state = sha2_compress_function(memory.get_range(ids.state, _sha256_state_size_felts), w)
+			//> segments.write_arg(ids.output, new_state)
+
+			Sha256InputChunkSize := uint64(16)
+
+			sha256Start, err := hinter.ResolveAsAddress(vm, sha256Start)
+			if err != nil {
+				return err
+			}
+
+			w, err := vm.Memory.GetConsecutiveMemoryValues(*sha256Start, Sha256InputChunkSize)
+			if err != nil {
+				return err
+			}
+
+			wUint32 := make([]uint32, len(w))
+			for i := 0; i < len(w); i++ {
+				value, err := hintrunnerUtils.ToSafeUint32(&w[i])
+				if err != nil {
+					return err
+				}
+				wUint32[i] = value
+			}
+
+			messageSchedule, err := utils.ComputeMessageSchedule(wUint32)
+			if err != nil {
+				return err
+			}
+
+			Sha256StateSize := uint64(8)
+
+			stateAddr, err := hinter.ResolveAsAddress(vm, state)
+			if err != nil {
+				return err
+			}
+
+			stateValuesMv, err := vm.Memory.GetConsecutiveMemoryValues(*stateAddr, Sha256StateSize)
+			if err != nil {
+				return err
+			}
+
+			var stateValues [8]uint32
+			for i := 0; i < len(stateValues); i++ {
+				value, err := hintrunnerUtils.ToSafeUint32(&stateValuesMv[i])
+				if err != nil {
+					return err
+				}
+				stateValues[i] = value
+			}
+
+			newState := utils.Sha256Compress(stateValues, messageSchedule)
+
+			output, err := hinter.ResolveAsAddress(vm, output)
+			if err != nil {
+				return err
+			}
+
+			for i := 0; i < len(newState); i++ {
+				newStateValue := mem.MemoryValueFromInt(newState[i])
+				outputOffset, err := output.AddOffset(int16(i))
```

Comment:
are you sure, that `i` will never exceed `int16` range? maybe just add `1` in each iteration?

Reply:
yeah i'm sure. `newState` will always be an array of 8 elements.